### DISCUSSION TITLE
Failed pings one way between nodes, not the other; help appreciated

### DISCUSSION THREAD

**Discussion by LordMike** [zwave-js/zwave-js#5179] [Q&A]
I'm looking at a situation where I cannot ping between nodes, in one direction - but I can in another. The [guide here](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/network-health) has no recommendations on `node -> node` pings, so I'm at a little bit of a loss as to what to do. Googling did not help either.

In my network, I have nodes `75`, `25` and `2` - the first two are both the same product, a ZDB5100 wall controller with built in dimmer and the last is a Siren from Aeotec. All three are within a few meters of each other, and all are mains powered devices. Using zwave-js-ui, I cannot ping in all directions between them, shown in the pictures below.

<details><summary>ZDB5100 to ZDB5100, 3 meters</summary>

Both ZDB5100's, in direct range of each other and each others neighbors. The curious here is that when I initiate from one end, it works in one direction, but initiating in the other end makes it fail both ways.

![image](https://user-images.githubusercontent.com/1027111/197705208-7f53c113-f49a-48ef-9175-03bdb139b41e.png)

![image](https://user-images.githubusercontent.com/1027111/197705657-393e26e2-1882-4db8-b5bc-1f2df01bb32f.png)

</details>

<details><summary>ZDB5100 to ZW162, 3 meters</summary>

In this case, I have a wall controller (75) and an Aeotec siren (2). In this case, messages from the siren (2) to the wall controller works but the other way does not - no matter which end initiates.

![image](https://user-images.githubusercontent.com/1027111/197706063-d566a59e-9de1-44d0-a723-92e58487cf16.png)

![image](https://user-images.githubusercontent.com/1027111/197706208-913d7c1a-bd1c-4de8-aafa-77cdffe0326a.png)

</details>

If this was a routed IP network, I would assume that some routes were misconfigured, ensuring that messages can flow in one direction but not the other. I know that there is routing capabilities in Z-wave, but I can't get to it - I can't see it anywhere.. The Z-wave PC Controller has a whole tab dedicated to routing, but I can't join my Gen5 stick to my Gen7 network (and I haven't had time to move the controller to my desktop for testing :P).. :/

Tips on how to resolve this would be very much appreciated :)

**Comment by AlCalzone** [comment]
I don't think the API to assign routes is documented, so you won't be able to use it for now.

That said, I think node 75 has truly terrible connectivity. While you were able to ping it from node 25, this only worked with a 4dB TX power reduction - at least 6dB are required for a stable connection.

**Reply by LordMike** [reply]
That’s just a physical limitation? 
Can I up the power (I think some devices have a config for it) somehow?

Write to the vendor and asks what they suggest? (It’s wall switches. so they’re physically confined)

**Reply by LordMike** [reply]
I just tried two of these that are facing each other in the same room. All pings come through, but at -1dBm and -2dBm. 

The score is 7/10

**Reply by LordMike** [reply]
So hypothetically I would need to get repeaters and place three in view (or at least better) than these wall switches - which are right now the powered backbone. 

.. or I’d have to find all the sweet routes. Where one switch on one side can bounce signals on a wall to reach another far away switch, and then relay to the switch that might be physically closer to the first, but has better connectivity with others. 

.. and then tune the network that way to ensure delivery.

**Reply by LordMike** [reply]
The problem here is that messages must sometimes get through, by chance, and then keep the route tables such that the messages use these suboptimal connections. 

I saw some writing about APR’s. Would this be a z-wave js job?

**Reply by LordMike** [reply]
Additional question - is Z-wave smart enough to send messages (in my network with one way being ok and the reverse being complete crap) through different routes, e.g. out one path and return another path?

**Reply by AlCalzone** [reply]
Yeah, nodes know multiple routes and will try them all, if the inbound route fails.

I plan to add some control over routes to zwave-js eventually.

**Reply by AlCalzone** [reply]
FWIW, APRs have been implemented in zwave-js. Now applications need to expose that functionality.
